aea/cli/common.py

```diff
@@ -183,8 +183,6 @@ def format_skills(items):
 def retrieve_details(name: str, loader: ConfigLoader, config_filepath: str):
     """Return description of a protocol, skill or connection."""
     config = loader.load(open(str(config_filepath)))
-    if config.name != name:
```

Comment:
Would this check plus a `raise AssertionError` not be preferable to the assertion? Then the user knows what's going on.

Reply:
I remove the if because I didn't want to write a test for that path. Also I don't feel this is the place to report an error like this back to "user land". We should have a "validity checker" for skills/connections/protocols which is run on skills as they are added to the registry and when the agent is run and report back to the user there.